src/embed/dispatch.rs

```diff
@@ -151,3 +156,30 @@ async fn do_debug(conns: &ConnMap, req: &Request) -> Response {
         _ => Err("Debug command not defined".into()),
     }
 }
+
+#[derive(Debug)]
+enum InitClientIdError {
+    GetErr(StoreError),
+    OpenErr(StoreError),
+    InvalidUtf8(std::string::FromUtf8Error),
+    PutClientIdErr(StoreError),
+    CommitErr(StoreError),
+}
+
+async fn init_client_id(s: &dyn Store) -> Result<String, InitClientIdError> {
+    use InitClientIdError::*;
+
+    const CID_KEY: &str = "sys/cid";
+    let cid = s.get(CID_KEY).await.map_err(GetErr)?;
+    if let Some(cid) = cid {
+        let s = String::from_utf8(cid).map_err(InvalidUtf8)?;
+        return Ok(s);
+    }
+    let wt = s.write().await.map_err(OpenErr)?;
```

Comment:
should probably test that it returns the same client id on a second call

Reply:
Done in next pr.